subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -104,16 +169,16 @@ int dfu_target_mcuboot_write(const void *const buf, size_t len)
 		LOG_ERR("flash_img_buffered_write error %d", err);
 		return err;
 	}
-
-	offset += len;
+	store_flash_img_context();
```

Comment:
I think you should check (and return) this value.

Reply:
nit: leave an emtpy line after the closing `}`

Reply:
Think I've fixed this now. I don't believe we should return an error as long as the write was good. Storing the progress is a nice to have but not storing it is not an critical problem.

Reply:
As you log any errors in `store_flash_img_context` that should be fine, but let's add a small comment here too, e.g. "failing to store progress is not a critical error".

Reply:
Added